spec/lib/mail_recipient_interceptor_spec.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+require "mail_recipient_interceptor"
+
+RSpec.describe MailRecipientInterceptor do
+  describe "#delivering_email" do
+    let(:original_recipient) { "not-allowed@example.com" }
+    let(:mail) { Mail.new(to: original_recipient) }
+
+    before do
+      ClimateControl.modify(GOVUK_NOTIFY_ALLOW_LIST: "allowed@example.com") do
```

Comment:
Would it be possible to use multiple email addresses here? I'm not sure how this env var should look to achieve the 'listiness' :-).

Reply:
Discussed with @emmabeynon - since the NotifyDeliveryMethod payload only supports a single email, we're going to remove the '_LIST' aspect of this PR, since we only anticipate needing one email address for the time being.